File: PS/JS/1149_greedy.js
let fs = require('fs');
let input = fs.readFileSync('./BJ/stdin').toString().split('\n');
let n = parseInt(input[0]);
let rgbcost = [];
for(let i = 1; i <= n; i++){
    let cur = input[i].split(' ');
    rgbcost.push( cur.map( (x)=>parseInt(x) ) );
}

function solution(rgbcost) {
    let answer = new Array(3).fill(0);
    for(let rgb of rgbcost){
        let newAnswer = new Array(3);
        for(let i = 0 ; i < 3 ; i++){
            newAnswer[i] = rgb[i] + Math.min(answer[(i+1)%3], answer[(i+2)%3]);
        }
        answer = newAnswer;
    }
    return Math.min(...answer);
}

console.log(solution(rgbcost));

File: PS/BJ/1149_greedy.cpp
#include <iostream>
#include <vector>
using namespace std;
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    int n;
    cin >> n;
    vector<vector<int>> rgbcost(n);
    for (int i = 0; i < n; i++) {
        int r, g, b;
        cin >> r >> g >> b;
        rgbcost[i].push_back(r);
        rgbcost[i].push_back(g);
        rgbcost[i].push_back(b);
    }
    vector<int> answer(3);
    for (vector<int> rgb : rgbcost) {
        vector<int> newanswer(3);
        for (int i = 0; i < 3; i++) {
            newanswer[i] = rgb[i] + min(answer[(i + 1) % 3], answer[(i + 2) % 3]);
        }
        answer = newanswer;
    }
    cout << min(answer[0], min(answer[1], answer[2]));
    return 0;
}
